insights/insights/doctype/insights_data_source/sources/base_database.py

```diff
@@ -83,26 +82,20 @@ def __init__(self):
         self.query_builder = None
         self.table_factory = None
 
-    def test_connection(self):
-        with self.connect() as connection:
+    def test_connection(self, log_errors=True):
+        with self.connect(log_errors=log_errors) as connection:
             res = connection.execute(text("SELECT 1"))
             return res.fetchone()
 
-    @retry(
```

Comment:
This was done to ignore the following error:
```
(pymysql.err.InternalError) Packet sequence number wrong - got 2 expected 3
(Background on this error at: https://sqlalche.me/e/20/2j85)
```

I assume `connect_timeout: 1` solves this issue?

Reply:
The `connect_timeout` kwarg is added to tackle the following problem: MySQL client making requests to a Postgres database doesn't immediately fail citing Database Type failure but rather connection timeout one after a hang timeout of ~10s, which is now bought down to 1s.

---

I haven't faced `(pymysql.err.InternalError) Packet sequence number wrong - got 2 expected 3` yet so I can't say if it solves it. However, I did some digging and it seems that this error comes up when PyMySQL is not used properly - particularly in the context of multithreading. _(someone also mentioned in one of the threads that the error went away after using `use_unicode` so can never be too sure 😄)_

Given we don't utilize pooling, the only other explanation I see is there were some issues with managing database sessions. Is this still a problem now? If so, we ought to fix this rather than applying wanton retries over the connect method.

---

Refs:
 - https://github.com/PyMySQL/PyMySQL/issues/423
 - https://github.com/tiangolo/fastapi/discussions/8852

Reply:
Will write the `retry` logic only for `mariadb` for now. Thanks for looking into it.